org.lflang/src/org/lflang/ASTUtils.java

```diff
@@ -769,77 +762,27 @@ public static <T> List<T> collectElements(Reactor definition, EStructuralFeature
     //// Utility functions for translating AST nodes into text
 
     /**
-     * Translate the given code into its textual representation.
-     * @param code AST node to render as string.
+     * Translate the given code into its textual representation
+     * with {@code CodeMap.Correspondence} tags inserted.
+     * @param node AST node to render as string.
      * @return Textual representation of the given argument.
      */
-    public static String toText(Code code) {
-        return CodeMap.Correspondence.tag(code, toUntaggedText(code), true);
+    public static String toText(EObject node) {
+        if (node == null)
+            return null;
```

Comment:
Should it be null or `""`? The latter seems safer.

Reply:
It seems non-obvious to me that the latter is safer. Is the AST node supposed to be null, or not? If it is OK for it to be null, then that should probably be documented. Since it's not documented, my best guess is that it's not OK to be null, in which case returning the empty string might be worse than doing nothing because converting `null` to `""` hides the original failure. We want to fail fast, right?

This is not the first time I have said something along these lines -- maybe we should discuss it to make sure I understand this design principle properly.

Reply:
I haven't looked at all the call sites, but in general, references to other AST nodes _could_ be null. I'm not sure what the right approach is either, but since this is used in code generation, returning an empty string seems relatively safe. Not sure though.